### PR TITLE
Added Recruitment Cycle Field

### PR DESCRIPTION
**Pull request by rohan-gupta8**
* Added initialisation data to populate
* Added an IntegerField called “recruiting_cycle” with 5 choices to club – Unknown, Fall, Spring, Both, Open
* Added to admin panel; included as filter option for admin panel
* Added to frontend filter logic
* Added to both clublist and club serializers

### CODE REVIEW

File: backend/clubs/migrations/0054_auto_20201101_1152.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                    (1, "Open Membership"),
+                    (3, "Audition Required"),
+                    (2, "Tryout Required"),
+                    (4, "Application Required"),
+                    (5, "Application and Interview Required"),
+                ],
+                default=4,
+            ),
+        ),
+        migrations.AlterField(
```
Comment (ezwang) [inline backend/clubs/migrations/0054_auto_20201101_1152.py:41]: These entries shouldn't appear in this commit. Perhaps you need to delete this commit, rebase with master, and regenerate this commit?

Comment (ezwang) [reply backend/clubs/migrations/0054_auto_20201101_1152.py:41]: Ah, never mind, I see that this is a typo fix. Nice!

**Review by ezwang — APPROVED** [review]
LGTM! Great work!

File: backend/clubs/migrations/0054_auto_20201101_1152.py (hunk 1 above)
Comment (rohan-gupta8) [reply backend/clubs/migrations/0054_auto_20201101_1152.py:41]: I changed a type in "Intereview" to "Interview".

Comment (rohan-gupta8) [reply backend/clubs/migrations/0054_auto_20201101_1152.py:41]: Is it still unintended?

**Pull request merged**